### ISSUE TITLE
Discussion about the classifier after the gradient reversal layer

### ISSUE THREAD

**Issue by oshindow** [StanislavParovoy/Multilingual_Text_to_Speech#1]
Should the output logits of the classifier be 0.5, 0.5, which followed by the gradient reversal layer, when there are 2 speakers, whose speaker id is 0 and 1, seperately? Therefore the cross entropy will be 0.6931, which can be lower. So the classifier would not tell different of those 2 speakers, if the object is to minimize the cross entropy. But the original intention we designed the gradient reversal layer is to remove speaker-specific information from the encoders. Isn't that the opposite of the goal?

Which means an ideal outcome doesn't correspond to a lower loss.
